### ISSUE TITLE
Make NEAR work for unfielded queries

### ISSUE THREAD

**Issue by aaccomazzi** [romanchyla/montysolr#173]
Right now the queries fail with a syntax error.  
Example: `x near2 y` or `x near y`

```
LUCENE_QUERY_CONVERSION_ERROR: Cannot convert query to lucene syntax: ConstantScore(author:x, author:x,*) error: (yet) Unsupported clause inside span query: org.apache.lucene.search.ConstantScoreQuery
```

**Comment by romanchyla** [comment]
also found another problem https://ui.adsabs.harvard.edu/search/q=%20author%3A%22Suyu%2C%20Sherry%20H.%22&sort=date%20desc%2C%20bibcode%20desc&p_=0

i'm somewhat puzzled, this used to work (before upgrade?)

ok, this should be the priority

**Comment by romanchyla** [comment]
fixed in 78efe39913c97152fe3caf8ab9920b5c2f54433d -- there were two problems; firstly, we are now wrapping some queries into a `constantquery` -- for those that should not produce score multiplication; there exists no equivalent for the span world, but I have solved it by setting boost factor to `0.0`

second problem is much deeper -- unfielded searches produce combinations of fields; proximity searches can however only work inside the same field; while `foo near2 bar` looks like could be solved by spreading the search into `title:foo near title:bar | bibcode:foo near bibcode:bar....` -- i have ultimately decided not to pursue this, because it is a path full of exceptions; for example when user makes a query `foo near bar near title:baz`

so I decided to generate a query exception whenever proximity conversion would result in a search across indexes
